### PR TITLE
Pin PyTorch-nightly version

### PR DESCRIPTION
**Pull request by Andrei-Aksionov**
Hi there 👋 

This project relies on the latest version of `PyTorch-nightly` which is fine, but what's not is that any issues with the nightly version affects this project too. 
It actually already happened to this user: https://github.com/Lightning-AI/lit-gpt/issues/327

**Proposal**: pin version of PyTorch-nightly and update it if necessary.

---

The version that I propose is already more than a month old, but it's the one that I've been using on a server with a GPU and it seems to work without any problems (though I haven't tested all the code).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,13 +88,13 @@ Luckily that is straightforward:
 **On CUDA**
 
 ```bash
-pip install --index-url https://download.pytorch.org/whl/nightly/cu118 --pre 'torch>=2.1.0dev'
+pip install --index-url https://download.pytorch.org/whl/nightly/cu118 --pre 'torch==2.1.0.dev20230703'
```
Comment (awaelchli) [inline README.md:91]: I don't remember how they handle it, but does this image persist? It might not be available in a few weeks, just saying :)

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Yeah actually, why don't we just take the latest (from yesterday)?

**Comment by Andrei-Aksionov** [comment]
So I've just checked the latest version of PyTorch-nightly which is `20230810`: looks like everything works except for `chat/base.py`: after sending a prompt it fails with exception:
```python
File ".../lit_gpt/model.py", line 273, in scaled_dot_product_attention
    return torch.nn.functional.scaled_dot_product_attention(
RuntimeError: cutlassF: no kernel found to launch!
```
And this problem has been around for at least 2 weeks. So maybe it's not a bug - it's a feature 😄.
What should we do: 

1. stick to the one month old version, maybe later fix this issue and update pinned PyTorch version
2. if the rest of the code relies on the newer version - pin the newest one and fix the issue in another PR

**Comment by carmocca** [comment]
I'd go with (2), working around the issue with https://github.com/Lightning-AI/lit-gpt/issues/327#issuecomment-1664674460.

**Comment by Andrei-Aksionov** [comment]
The original reason for this PR was issue #327: after updating PyTorch chat/base.py stopped working and on the earlier version of torch it worked fine.
The problem was in the wrongly provided precision: by default bf16 was used for all cards, even for T4, which doesn't support it.
On previous versions of torch it didn't cause any issues for some reason.

So, since the issue that let to this PR is fixed, then maybe we can drop this PR and continue using the latest and greatest version of torch? At least until next issue ;)

**Comment by carmocca** [comment]
We can close this issue. But we should soon pin to the PyTorch 2.1 release. Which had a first release candidate published this week

**Pull request closed without merging**